crates/ruff/src/rules/ruff/rules/implicit_optional.rs

```diff
@@ -333,15 +336,42 @@ pub(crate) fn implicit_optional(checker: &mut Checker, arguments: &Arguments) {
         let Some(annotation) = &def.annotation else {
             continue
         };
-        let Some(expr) = type_hint_explicitly_allows_none(annotation, checker.semantic()) else {
-            continue;
-        };
-        let conversion_type = checker.settings.target_version.into();
 
-        let mut diagnostic = Diagnostic::new(ImplicitOptional { conversion_type }, expr.range());
-        if checker.patch(diagnostic.kind.rule()) {
-            diagnostic.try_set_fix(|| generate_fix(checker, conversion_type, expr));
+        if let Expr::Constant(ast::ExprConstant {
+            range,
+            value: Constant::Str(value),
+            kind: _,
+        }) = annotation.as_ref()
```

Comment:
nit: unused and might be confused with annotation kind down below

```suggestion
        if let Expr::Constant(ast::ExprConstant {
            range,
            value: Constant::Str(value),
            ..
        }) = annotation.as_ref()
```